checker/manual/creating-a-checker.tex

```diff
@@ -1094,31 +1090,94 @@ \section{The checker class:  Compiler interface\label{writing-compiler-interface
+called \<qual> located in the same directory as the checker, or by overriding
 the
 \refmethodterse{framework/type}{AnnotatedTypeFactory}{createSupportedTypeQualifiers}{--}
-method.  Each argument to \<@TypeQualifiers> or value returned by
-\<createSupportedTypeQualifiers> is a class literal for a type qualifier
-whose definition bears the
-\refqualclass{framework/qual}{TypeQualifier} meta-annotation.
+method. By default, the polymorphic qualifier
+\refqualclass{framework/qual}{PolyAll} is supported by every checker. It can be
+optionally excluded.
+
+There are four recommended ways to write an overriding implementation of
+\refmethodterse{framework/type}{AnnotatedTypeFactory}{createSupportedTypeQualifiers}{--}:
+
+\begin{enumerate}
+\item using reflective lookup to add all qualifiers, and exclude
+\code{@PolyAll}. Example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return loadTypeQualifiersFromQualDir(true, null);
+}
+\end{Verbatim}
+
+\item using reflective lookup to add qualifiers, exclude \code{@PolyAll}, and
+add a set of qualifiers. Example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return loadTypeQualifiersFromQualDir(true,
+        new HashSet<Class<? extends Annotation>>(
+            Arrays.asList(H1Top.class, H1S1.class, H1S2.class,
+                H1Bot.class, H2Top.class, H2S1.class, H2S2.class,
+                H2Bot.class, H1Poly.class, H2Poly.class)));
+}
+\end{Verbatim}
+
+The use of a set is necessary for any qualifiers that are not located in
+the \<qual> folder, or for qualifiers that do not or cannot have the
+\sunjavadocanno{java/lang/annotation/Target.html}{Target} meta-annotation with
+the value \<ElementType.TYPE\_USE> and/or \<ElementType.TYPE\_PARAMETER> (e.g.,
+qualifiers that should not be directly used in source code, but are still part
+of the hierarchy).
+
+\item using reflective lookup to add qualifiers, include \code{@PolyAll}, and
+add a set of qualifiers. Example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return loadTypeQualifiersFromQualDir(false,
+        new HashSet<Class<? extends Annotation>>(
+            Arrays.asList(IGJBottom.class)));
+}
+\end{Verbatim}
+
+\item only using a set of qualifiers. Example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return Collections.unmodifiableSet(
+        new HashSet<Class<? extends Annotation>>(
+            Arrays.asList(A.class, B.class, C.class, D.class, E.class, F.class)));
+}
+
+The set of qualifiers returned by \code{createSupportedTypeQualifiers()} must be
+an unmodifiable set.
+
+\end{Verbatim}
+
+\end{enumerate}
+
+Each value returned by \<createSupportedTypeQualifiers> is an instance of an Annotation Class of a type qualifier
```

Comment:
Please rephrase, because each value returned by createSupportedTypeQualifiers is a Set, not an instance of an annotation class.

Also, why is "Annotation Class" capitalized?